scripts/create_dataset.py

```diff
@@ -1,69 +1,161 @@
+    :param column_datatype: the primitive type enforced for values of the input JSON column
+    :param column_array_of: indicates if the column values should be in array
+    :param input_table_name: the name of input JSON table that has the input_column
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    relationships = [
+        {
+            "name": "{}_{}_to_{}_{}".format(input_table_name,
+                                            column_name,
+                                            input_relationship["table_name"],
+                                            input_relationship["column_name"]),
+            "from": {"table": input_table_name, "column": column_name, "cardinality": "one"},
+            "to": {
+                    "table": input_relationship["table_name"],
+                    "column": input_relationship["column_name"],
+                    "cardinality": "many" if column_array_of is True else "one"
+                  }
+        }
+        if validate_table_and_column(
+               table_name=input_table_name,
+               column_name=column_name,
+               input_tables=input_tables) and validate_table_and_column(table_name=input_relationship["table_name"],
+                                                                        column_name=input_relationship["column_name"],
+                                                                        input_tables=input_tables)
+        else []
+        for input_relationship in input_relationships] if input_relationships else []
+
+    column = {
+        "name": column_name,
+        "datatype": "string" if column_datatype == "link" else column_datatype,
+        "array_of": column_array_of is True
+    }
+
+    return column, relationships
+
+
+def validate_table_and_column(table_name, column_name, input_tables):
+    """
+    Validates a relationship checking the the given table and column exist in input tables.
+    :param table_name: the name of table containing has the input_column that will be validated against the input tables
+    :param column_name: the name of the input_column that will be validated against the input tables
+    :param input_tables: the input tables that have been added to a dict and used to validate the created relationships
+    """
+    if True in [
+        True in [
+                input_column["name"] == column_name
+                for input_column in input_tables[input_table_name]["columns"]]
+            if table_name == input_table_name else [] for input_table_name in input_tables]:
+        return True
+
+    # raise in error if either the column or table name does not exist in the input tables
+    sys.stderr.write("\nThe table name and relationship: Column, {}, in table, {} does not exist\n"
+                     .format(column_name, table_name))
+    raise SystemExit(errno.EBADF)
```

Comment:
Wot this errno?

Reply:
Instead of having this method throw, let's have it return `False`. The caller in main code can still throw the error, but it'll make testing this method a lot cleaner.